module-system/sov-modules-macros/src/runtime_event.rs

```diff
@@ -0,0 +1,115 @@
+use proc_macro2::Span;
+use syn::DeriveInput;
+
+use super::common::{
+    get_generics_type_param, get_serialization_attrs, StructDef, StructFieldExtractor,
+};
+
+pub(crate) const EVENT: &str = "Event";
+
+pub(crate) struct RuntimeEventMacro {
+    field_extractor: StructFieldExtractor,
+}
+
+impl<'a> StructDef<'a> {
+    fn create_event_enum_legs(&self) -> Vec<proc_macro2::TokenStream> {
+        self.fields
+            .iter()
+            .map(|field| {
+                let name = &field.ident;
+                let ty = &field.ty;
+
+                quote::quote!(
+                    #[doc = "Module event."]
```

Comment:
nit: This doc is a little terse. Maybe we could say something like, "an event emitted by the {} module"?

Reply:
Great catch, this was from previous PR, but will improve it with your suggestion.